src/nvim/misc1.c

```diff
@@ -1313,30 +1304,29 @@ int plines_win_col(win_T *wp, linenr_T lnum, long column)
   if (wp->w_width == 0)
     return lines + 1;
 
-  line = s = ml_get_buf(wp->w_buffer, lnum, FALSE);
+  char_u *line = ml_get_buf(wp->w_buffer, lnum, false);
+  char_u *s = line;
 
-  col = 0;
+  colnr_T col = 0;
   while (*s != NUL && --column >= 0) {
-    col += win_lbr_chartabsize(wp, line, s, (colnr_T)col, NULL);
+    col += win_lbr_chartabsize(wp, line, s, col, NULL);
     mb_ptr_adv(s);
   }
 
-  /*
-   * If *s is a TAB, and the TAB is not displayed as ^I, and we're not in
-   * INSERT mode, then col must be adjusted so that it represents the last
-   * screen position of the TAB.  This only fixes an error when the TAB wraps
-   * from one screen line to the next (when 'columns' is not a multiple of
-   * 'ts') -- webb.
-   */
-  if (*s == TAB && (State & NORMAL) && (!wp->w_p_list || lcs_tab1))
-    col += win_lbr_chartabsize(wp, line, s, (colnr_T)col, NULL) - 1;
+  // If *s is a TAB, and the TAB is not displayed as ^I, and we're not in
+  // INSERT mode, then col must be adjusted so that it represents the last
+  // screen position of the TAB.  This only fixes an error when the TAB wraps
+  // from one screen line to the next (when 'columns' is not a multiple of
+  // 'ts') -- webb.
+  if (*s == TAB && (State & NORMAL) && (!wp->w_p_list || lcs_tab1)) {
+    col += win_lbr_chartabsize(wp, line, s, col, NULL) - 1;
+  }
 
-  /*
-   * Add column offset for 'number', 'relativenumber', 'foldcolumn', etc.
-   */
-  width = wp->w_width - win_col_off(wp);
-  if (width <= 0)
+  // Add column offset for 'number', 'relativenumber', 'foldcolumn', etc.
+  int width = wp->w_width - win_col_off(wp);
+  if (width <= 0) {
     return 9999;
```

Comment:
Hmm, this almost makes sense, but the max number of lines is 1000 not 10000. (or maybe it just says the number of lines IS OVER 9000)